index.js

```diff
@@ -393,6 +385,27 @@ function maybeRetryMS(error) {
     }
   }
 
+  // If the retry limit has been reached, abort the request
+  if (computedValue == 0 && retryOptions.limit > 0) {
```

Comment:
```suggestion
  if (computedValue === 0) {
```

`retryOptions.limit` is guaranteed to be bigger than zero, otherwise `maybeRetryMS()` is not called.